src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private String fruitName;
+    private int quantity;
+    private Operation operation;
+
+    public FruitTransaction(String fruitName, int quantity, Operation operation) {
+        this.fruitName = fruitName;
+        this.quantity = quantity;
+        this.operation = operation;
+    }
+
+    public String getFruitName() {
+        return fruitName;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;

```

Comment:
The 'code' field inside the 'Operation' enum should be private to follow the encapsulation principle.